samcli/hook_packages/terraform/hooks.py

```diff
@@ -0,0 +1,357 @@
+"""
+Terraform hooks implementation module
+"""
+
+from dataclasses import dataclass
+import json
+import os
+from subprocess import run, CalledProcessError
+from tempfile import NamedTemporaryFile
+from typing import Any, Callable, Dict, List, Optional, Union
+import hashlib
+import logging
+
+from samcli.lib.hook.exceptions import PrepareHookException
+from samcli.lib.utils.hash import str_checksum
+from samcli.lib.utils.resources import (
+    AWS_LAMBDA_FUNCTION as CFN_AWS_LAMBDA_FUNCTION,
+)
+
+LOG = logging.getLogger(__name__)
+
+TF_AWS_LAMBDA_FUNCTION = "aws_lambda_function"
+PROVIDER_NAME = "registry.terraform.io/hashicorp/aws"
+
+# max logical id len is 255
+LOGICAL_ID_HASH_LEN = 8
+LOGICAL_ID_MAX_HUMAN_LEN = 247
+
+PropertyBuilder = Callable[[dict], Any]
+PropertyBuilderMapping = Dict[str, PropertyBuilder]
+
+
+@dataclass
+class ResourceTranslator:
+    cfn_name: str
+    property_builder_mapping: PropertyBuilderMapping
+
+
+def prepare(params: dict) -> dict:
+    """
+    Prepares a terraform application for use with the SAM CLI
+
+    Parameters
+    ----------
+    params: dict
+        Parameters of the IaC application
+
+    Returns
+    -------
+    dict
+        information of the generated metadata files
+    """
+    output_dir_path = params.get("OutputDirPath")
+    if not output_dir_path:
+        raise PrepareHookException("OutputDirPath was not supplied")
+
+    try:
+        # initialize terraform application
```

Comment:
can you add debugging message instead of these comments or adding to it :)

Reply:
my point is TF commands usually take time, so may be it is something good to let the customers what we are doing right now, so they do not think that there is an issue. Also, if possible to write the commands output and error streams for debugging, and specially in case of failures

Reply:
done